### PR TITLE
[guilib] - listen to GUI_MSG_SETFOCUS for all container types

### PR DESCRIPTION
**Pull request by phil65**
@xhaggi @mkortstiege
this  allows Control.SetFocus(CONTAINER_ID,ITEM_INDEX) for all container types.

### CODE REVIEW

**Comment by phil65** [comment]
hmm or should we handle this in CGUIBaseContainer?

File: xbmc/guilib/GUIPanelContainer.cpp [hunk 1]
```diff
@@ -254,6 +254,14 @@ bool CGUIPanelContainer::OnMessage(CGUIMessage& message)
       SetCursor(0);
       // fall through to base class
     }
+    else if (message.GetMessage() == GUI_MSG_SETFOCUS)
+    {
+        if (message.GetParam1()) // subfocus item is specified, so set the offset appropriately
+        {
+            int item = std::min(GetOffset() + (int)message.GetParam1() - 1, (int)m_items.size() - 1);
```
Comment (xhaggi) [inline xbmc/guilib/GUIPanelContainer.cpp]: `message.GetParam1()` returns an int so there is no need for the cast.

and please use c++ style casts for `m_items.size()` -> `static_cast<int>(m_items.size() - 1)`

**Comment by phil65** [comment]
Question still stands: Should we deal with that in parent class or in every container class itself?

**Comment by xhaggi** [comment]
IMO it makes sense to put this into `CGUIBaseContainer and remove it from `CGUIListContainer::OnMessage()`

**Comment by phil65** [comment]
updated, I´m just movin the code part for GUI_MSG_SETFOCUS  to the base class now.

**Comment by mkortstiege** [comment]
Indentation is wrong, other than that it looks good to me.

**Comment by phil65** [comment]
@mkortstiege fixed indent.

**Comment by mkortstiege** [comment]
Thanks, i guess this is J* as well? jenkins build this please.

**Comment by phil65** [comment]
yep, definitely J*.

**Comment by phil65** [comment]
jenkins build this please.

**Comment by phil65** [comment]
@ronie bump for skin changes thread.

**Pull request merged**